common/src/main/java/io/pravega/common/concurrent/MultiKeyLatestItemSequentialProcessor.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.concurrent;
+
+import com.google.common.base.Preconditions;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+
+/**
+ * Provides a way to run a function on Key Value Pairs but guarantees that the function is only invoked for one item at a time per key.
+ * If multiple 'updates' to a particular key  are provided while the processor is running, only the most recent is passed to the
+ * function.
+ * This allows it to 'skip' updates for keys that are not the most recent value.
+ *
+ * @param <KeyType> The type of the key to be processed.
+ * @param <ItemType> The type of item to be processed.
+ */
+public class MultiKeyLatestItemSequentialProcessor<KeyType, ItemType> {
```

Comment:
I strongly recommend you make this class implement `AutoCloseable` and add a `close` method that would cancel any ongoing executions and prevent new items from being added. Helps a lot with graceful shutdowns.

Reply:
At present, the lifecycle of the executor is managed by the `io.pravega.shared.metrics.ClientMetricUpdater` which instantiates an instance of `MultiKeyLatestItemSequentialProcessor`. 
Also, `io.pravega.shared.metrics.MetricNotifier` implements `AutoCloseable` .